packages/charts/react-charting/src/components/DonutChart/DonutChart.base.tsx

```diff
@@ -324,15 +338,21 @@ export class DonutChartBase extends React.Component<IDonutChartProps, IDonutChar
   };
 
   private _valueInsideDonut(valueInsideDonut: string | number | undefined, data: IChartDataPoint[]) {
-    const highlightedLegend = this._getHighlightedLegend();
-    if (valueInsideDonut !== undefined && (highlightedLegend !== '' || this.state.showHover)) {
+    const highlightedLegends = this._getHighlightedLegend();
+    if (valueInsideDonut !== undefined && (highlightedLegends.length > 0 || this.state.showHover)) {
       let legendValue = valueInsideDonut;
+      let totalValue = 0;
       data!.map((point: IChartDataPoint, index: number) => {
-        if (point.legend === highlightedLegend || (this.state.showHover && point.legend === this.state.legend)) {
+        if (
+          highlightedLegends.includes(point.legend!) ||
+          (this.state.showHover && point.legend === this.state.legend)
```

Comment:
> point.legend === this.state.legend [](http://example.com/codeflow?start=34&length=35)

you are adding selected legend and hover legend values together?

Reply:
separated the logic